com.unity.render-pipelines.high-definition/Editor/VFXGraph/Outputs/VFXAbstractParticleHDRPLitOutput.cs

```diff
@@ -431,13 +431,13 @@ public override void OnEnable()
                 switch (blendMode)
                 {
                     case BlendMode.Alpha:
-                        forwardDefines.WriteLine("#define _BLENDMODE_ALPHA");
+                        forwardDefines.WriteLine("#define _BlendMode 0");
```

Comment:
hmm... wondering if we can define based on 

#define BLENDMODE_ALPHA (0)
#define BLENDMODE_PREMULTIPLY (4)
#define BLENDMODE_ADDITIVE (1)

here

Reply:
No won't work as we don't have the include in hlsl all the time for vfx last I checked. I can see if I can make sure that is included, but there must have been a reason.

Reply:
That said, while def looks better, it is still hardcoded in a string, so won't save our ass in case of changes.

Reply:
FYI, this defines lands here
Unlit :
![image](https://user-images.githubusercontent.com/22494706/94893026-7df0e480-0486-11eb-9654-cf749c4bb72c.png)

Lit : 
![image](https://user-images.githubusercontent.com/22494706/94893247-01aad100-0487-11eb-934f-22a3b5bfb706.png)

Indeed, there isn't any include before these define, we can potential feed these `#define BLENDMODE_ALPHA (0)` within the same function, before the switch, not sure it's worth it.
Isn't there a enum which fits these value ? I could be a generated string.
```
#define BLENDMODE_ALPHA (0)
#define BLENDMODE_PREMULTIPLY (4)
#define BLENDMODE_ADDITIVE (1)
```

@FrancescoC-unity Can you confirm these defines are only needed for lit output ? The output which are including :
- Packages/com.unity.render-pipelines.high-definition/Runtime/Material/Material.hlsl
- Packages/com.unity.render-pipelines.high-definition/Runtime/Material/Lit/Lit.hlsl
- Packages/com.unity.render-pipelines.high-definition/Runtime/Material/BuiltinUtilities.hlsl

Reply:
There is an enum on C# that generate an hlsl file with the defines, that hlsl file is included in Material/Material.hlsl
 , **however** we cannot do our define after the include of Material.hlsl, otherwise it won't compile. So we need an additional include for that specific file. 

Whether is worth it or not I guess I'll leave the decision to you guys, feels a bit overkill to me, but indeed it'd make it a tad more readable. 

This would be needed for whatever includes Material.hlsl, so yes, I assume only lit output.

Reply:
no, it is required for both lit and unlit transparent.
ok nevermind for now, this is something that can't change anyway to not break the universe, let's just add a comment on which enum it map for now.

Reply:
No, unlit vfx doesn't use the blend mode as in HDRP, so does not need this as it does not include Material.hlsl

Reply:
This was verified btw running the test, if it was needed for Unlit too the test wouldn't have compiled

Reply:
> no, it is required for both lit and unlit transparent.

> No, unlit vfx doesn't use the blend mode as in HDRP,

I confirm @FrancescoC-unity point, only needed for Lit HDR. For instance, it's needed using internal EvaluateAtmosphericScattering
For the fog evaluation, we are manually applying falloff using other defines : https://github.cds.internal.unity3d.com/unity/vfx-graphics/blob/vfx/staging/com.unity.render-pipelines.high-definition/Runtime/VFXGraph/Shaders/VFXCommon.hlsl#L154

If there is a compilation error in case of mistake, that's fine !